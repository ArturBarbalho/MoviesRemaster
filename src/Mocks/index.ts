import { BannerProps, CardProps, bigbannerProps } from "../typesProps";
export const BiggerBannerMock:bigbannerProps = {
  bigsrc: 'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
  smallsrc:'https://image.tmdb.org/t/p/original//ifRFLx83Xk1DcwAS3OScgI6HmWO.jpg',
  genres:['Action', 'Comedy'],
  id:2,
  media:'movie',
  overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
  title:'ONYANKOPON 3 O IN√çCIO'
}
export const CardMock:CardProps = {
  id:2,
  media:'movie',
  src:'http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg',
  title:'ONYANKOPON 2 O RETORNO'
}

export const BannerMock:BannerProps = {
    media: 'movie',
    relaseDate: '(1800)',
    id:1,
    title:'ONYANKOPON',
    src:'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
    overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
    genres: ['Action', 'Comedy' ]
  }