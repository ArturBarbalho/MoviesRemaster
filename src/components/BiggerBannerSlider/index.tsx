import { settings } from '../../sliderSettings/biggerbanner'
import { bigbannerProps } from '../../typesProps'
import BiggerBanner from '../BiggerBanner'
import Slider from '../Slider'
import * as S from './styles'
type props = {
  props: bigbannerProps[]
}
const BiggerBannerSlider = ({props}:props) =>{
const slide = props.map((e)=>{
  return <BiggerBanner key={e.title} props={e} />
})
return(
  <S.Wrapper>
    <Slider settings={settings}> {slide} </Slider>
  </S.Wrapper>
)
}
export default BiggerBannerSlider