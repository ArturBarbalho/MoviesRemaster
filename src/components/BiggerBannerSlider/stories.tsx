import { Story, Meta } from '@storybook/react/types-6-0'
import BiggerBannerSlider from '.'
import { BiggerBannerMock } from '../../Mocks/index';

export default {
  title: 'BiggerBannerSlider',
  component: BiggerBannerSlider
} as Meta
const mock = [BiggerBannerMock,BiggerBannerMock,BiggerBannerMock]
export const Default: Story = () => <BiggerBannerSlider props={mock}/>