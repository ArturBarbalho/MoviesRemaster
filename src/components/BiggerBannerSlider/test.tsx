import { render, screen } from '@testing-library/react'
import '../../../.jest/match-media-mock'
import BiggerBannerSlider from '.'
import { BiggerBannerMock } from '../../Mocks'

const mock = [BiggerBannerMock,BiggerBannerMock]

describe('<BiggerBannerSlider />', () => {
  it('should render the heading', () => {
    const { container } = render(<BiggerBannerSlider props={mock} />)

    expect(container.querySelectorAll('.slick-active')).toHaveLength(1)

    expect(container.firstChild).toMatchSnapshot()
  })
})