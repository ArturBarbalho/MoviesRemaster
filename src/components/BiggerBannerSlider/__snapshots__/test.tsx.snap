// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BiggerBannerSlider /> should render the heading 1`] = `
<section
  class="styles__Wrapper-sc-ks0hj3-0"
>
  <section
    class="styles__Wrapper-sc-12nwagl-0 fTUMGo"
  >
    <div
      class="slick-slider slick-initialized"
      dir="ltr"
    >
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-1"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-18y4bwb-0"
              >
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 gHPgtQ"
                  src="https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg"
                />
                <div
                  class="styles__Cont-sc-18y4bwb-1 kgmLvR"
                  data-testid="cont"
                >
                  <img
                    class="styles__Wrapper-sc-y6e3xk-0 cePpyH"
                    src="https://image.tmdb.org/t/p/original//ifRFLx83Xk1DcwAS3OScgI6HmWO.jpg"
                  />
                  <div>
                    <div
                      class="styles__Wrapper-sc-6y5ihb-1 daGpDx"
                    >
                      <div
                        class="styles__Genre-sc-6y5ihb-0 egsKaq"
                      >
                        Action
                      </div>
                      <div
                        class="styles__Genre-sc-6y5ihb-0 egsKaq"
                      >
                        Comedy
                      </div>
                    </div>
                    <p
                      class="styles__Wrapper-sc-mhbed0-0 gpZHFY"
                    >
                      Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...
                    </p>
                    <div
                      class="styles__ButtonDiv-sc-18y4bwb-2 brWXpG"
                    >
                      <button
                        class="styles__Wrapper-sc-11qq8g6-0 jDrFTe"
                        color="gray"
                      >
                        See more
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-18y4bwb-0"
              >
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 gHPgtQ"
                  src="https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg"
                />
                <div
                  class="styles__Cont-sc-18y4bwb-1 kgmLvR"
                  data-testid="cont"
                >
                  <img
                    class="styles__Wrapper-sc-y6e3xk-0 cePpyH"
                    src="https://image.tmdb.org/t/p/original//ifRFLx83Xk1DcwAS3OScgI6HmWO.jpg"
                  />
                  <div>
                    <div
                      class="styles__Wrapper-sc-6y5ihb-1 daGpDx"
                    >
                      <div
                        class="styles__Genre-sc-6y5ihb-0 egsKaq"
                      >
                        Action
                      </div>
                      <div
                        class="styles__Genre-sc-6y5ihb-0 egsKaq"
                      >
                        Comedy
                      </div>
                    </div>
                    <p
                      class="styles__Wrapper-sc-mhbed0-0 gpZHFY"
                    >
                      Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...
                    </p>
                    <div
                      class="styles__ButtonDiv-sc-18y4bwb-2 brWXpG"
                    >
                      <button
                        class="styles__Wrapper-sc-11qq8g6-0 jDrFTe"
                        color="gray"
                      >
                        See more
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-18y4bwb-0"
              >
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 gHPgtQ"
                  src="https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg"
                />
                <div
                  class="styles__Cont-sc-18y4bwb-1 kgmLvR"
                  data-testid="cont"
                >
                  <img
                    class="styles__Wrapper-sc-y6e3xk-0 cePpyH"
                    src="https://image.tmdb.org/t/p/original//ifRFLx83Xk1DcwAS3OScgI6HmWO.jpg"
                  />
                  <div>
                    <div
                      class="styles__Wrapper-sc-6y5ihb-1 daGpDx"
                    >
                      <div
                        class="styles__Genre-sc-6y5ihb-0 egsKaq"
                      >
                        Action
                      </div>
                      <div
                        class="styles__Genre-sc-6y5ihb-0 egsKaq"
                      >
                        Comedy
                      </div>
                    </div>
                    <p
                      class="styles__Wrapper-sc-mhbed0-0 gpZHFY"
                    >
                      Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...
                    </p>
                    <div
                      class="styles__ButtonDiv-sc-18y4bwb-2 brWXpG"
                    >
                      <button
                        class="styles__Wrapper-sc-11qq8g6-0 jDrFTe"
                        color="gray"
                      >
                        See more
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="2"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-18y4bwb-0"
              >
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 gHPgtQ"
                  src="https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg"
                />
                <div
                  class="styles__Cont-sc-18y4bwb-1 kgmLvR"
                  data-testid="cont"
                >
                  <img
                    class="styles__Wrapper-sc-y6e3xk-0 cePpyH"
                    src="https://image.tmdb.org/t/p/original//ifRFLx83Xk1DcwAS3OScgI6HmWO.jpg"
                  />
                  <div>
                    <div
                      class="styles__Wrapper-sc-6y5ihb-1 daGpDx"
                    >
                      <div
                        class="styles__Genre-sc-6y5ihb-0 egsKaq"
                      >
                        Action
                      </div>
                      <div
                        class="styles__Genre-sc-6y5ihb-0 egsKaq"
                      >
                        Comedy
                      </div>
                    </div>
                    <p
                      class="styles__Wrapper-sc-mhbed0-0 gpZHFY"
                    >
                      Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...
                    </p>
                    <div
                      class="styles__ButtonDiv-sc-18y4bwb-2 brWXpG"
                    >
                      <button
                        class="styles__Wrapper-sc-11qq8g6-0 jDrFTe"
                        color="gray"
                      >
                        See more
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="3"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-18y4bwb-0"
              >
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 gHPgtQ"
                  src="https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg"
                />
                <div
                  class="styles__Cont-sc-18y4bwb-1 kgmLvR"
                  data-testid="cont"
                >
                  <img
                    class="styles__Wrapper-sc-y6e3xk-0 cePpyH"
                    src="https://image.tmdb.org/t/p/original//ifRFLx83Xk1DcwAS3OScgI6HmWO.jpg"
                  />
                  <div>
                    <div
                      class="styles__Wrapper-sc-6y5ihb-1 daGpDx"
                    >
                      <div
                        class="styles__Genre-sc-6y5ihb-0 egsKaq"
                      >
                        Action
                      </div>
                      <div
                        class="styles__Genre-sc-6y5ihb-0 egsKaq"
                      >
                        Comedy
                      </div>
                    </div>
                    <p
                      class="styles__Wrapper-sc-mhbed0-0 gpZHFY"
                    >
                      Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...
                    </p>
                    <div
                      class="styles__ButtonDiv-sc-18y4bwb-2 brWXpG"
                    >
                      <button
                        class="styles__Wrapper-sc-11qq8g6-0 jDrFTe"
                        color="gray"
                      >
                        See more
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</section>
`;
